### ISSUE TITLE
Git checkout v0.8.2 cannot install Dart SDK.3 error

### ISSUE THREAD

**Issue by chadidi** [flutter/flutter#22961]
## Steps to Reproduce

1. `git checkout v0.8.2`
2. `flutter packages get` on any application

## Logs
```
$ flutter packages get
Downloading Dart SDK from Flutter engine 58a1894a1cd798de2f9a206f157a90d45944028b...
Unknown operating system. Cannot install Dart SDK.3
```

```
$ flutter --verbose create foo
Building flutter tool...
/c/src/flutter/bin/flutter: line 46: /c/src/flutter/bin/cache/dart-sdk/bin/pub:No such file or directory
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)
/c/src/flutter/bin/flutter: line 46: /c/src/flutter/bin/cache/dart-sdk/bin/pub:No such file or directory
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (8 tries left)

```

```
$ flutter doctor -v
Building flutter tool...
/c/src/flutter/bin/flutter: line 46: /c/src/flutter/bin/cache/dart-sdk/bin/pub:No such file or directory
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)
/c/src/flutter/bin/flutter: line 46: /c/src/flutter/bin/cache/dart-sdk/bin/pub:No such file or directory
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (8 tries left)
```

**Comment by zoechi** [comment]
Please try 
```
git clean -xfd
git pull
flutter doctor
```

**Comment by chadidi** [comment]
@zoechi the solution above didn't work I had to delete the entire flutter SDK folder and extract it again. 

and that fixed #22940 #22928 too.

**Comment by zoechi** [comment]
Glad to hear you could make it work.
So is it ok to close all 3 issues then?
